compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1340,7 +1341,15 @@ TR::Instruction *OMR::Power::MemoryReference::expandInstruction(TR::Instruction
 
          if (displacement < LOWER_IMMED || displacement > UPPER_IMMED)
             {
-            prevInstruction = generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, base, base, cg->hiValue(displacement), prevInstruction);
+            if (0x00008000 == cg->hiValue(displacement))
+               {
+               generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, base, base, 0x7FFF);
+               generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, base, base, 0x1);
+               }
+            else
+               {
+               prevInstruction = generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, base, base, cg->hiValue(displacement), prevInstruction);
+               }
```

Comment:
I'd like to see test cases added to the unit tests for MemoryReference expansion to ensure that this code works correctly and to make sure we don't regress at some point in the future.